example/heatEquation2D/src/heatEquation2D.cpp

```diff
@@ -0,0 +1,194 @@
+    BufAcc uCurrBufAcc{alpaka::allocBuf<double, Idx>(devAcc, extent)};
+
+    // Set buffer to initial conditions
+    initalizeBuffer(uBufHost, extent, dx, dy);
+
+    // Select queue
+    using QueueProperty = alpaka::NonBlocking;
+    using QueueAcc = alpaka::Queue<Acc, QueueProperty>;
+    QueueAcc queue1{devAcc};
+    QueueAcc queue2{devAcc};
+
+    // Copy host -> device
+    alpaka::memcpy(queue1, uCurrBufAcc, uBufHost);
+    alpaka::wait(queue1);
+
+    // Define a workdiv for the given problem
+    constexpr alpaka::Vec<Dim, Idx> elemPerThread{1, 1};
+    // Appropriate chunk size to split your problem for your Acc
+    constexpr alpaka::Vec<Dim, Idx> chunkSize{16u, 16u};
+    constexpr alpaka::Vec<Dim, Idx> chunkSizeWithHalo{chunkSize[0] + haloSize[0], chunkSize[1] + haloSize[1]};
+
+    auto const maxThreadsPerBlock = alpaka::getAccDevProps<Acc>(devAcc).m_blockThreadExtentMax;
+    auto const threadsPerBlock = maxThreadsPerBlock.prod() < chunkSize.prod() ? maxThreadsPerBlock : chunkSize;
+
+    constexpr alpaka::Vec<Dim, Idx> numChunks{
+        (extent[0] - haloSize[0] - 1) / chunkSize[0] + 1,
+        (extent[1] - haloSize[1] - 1) / chunkSize[1] + 1};
+
+    static_assert(
+        (extent[0] - haloSize[0]) % chunkSize[0] == 0 && (extent[1] - haloSize[1]) % chunkSize[1] == 0,
+        "Domain must be divisible by chunk size");
+
+    auto const pitchCurrAcc{alpaka::getPitchesInBytes(uCurrBufAcc)};
+
+    alpaka::WorkDivMembers<Dim, Idx> workDiv_manual{numChunks, threadsPerBlock, elemPerThread};
+
+    StencilKernel<chunkSizeWithHalo.prod()> stencilKernel;
+    BoundaryKernel boundaryKernel;
+
+    for(uint32_t step = 1; step <= numTimeSteps; ++step)
+    {
+        // Compute next values
+        alpaka::exec<Acc>(
+            queue1,
+            workDiv_manual,
+            stencilKernel,
+            uCurrBufAcc.data(),
+            uNextBufAcc.data(),
+            chunkSize,
+            pitchCurrAcc,
+            dx,
+            dy,
+            dt);
+
+        // apply boundaries
+        alpaka::exec<Acc>(
+            queue1,
+            workDiv_manual,
+            boundaryKernel,
+            uNextBufAcc.data(),
+            chunkSize,
+            pitchCurrAcc,
+            step,
+            dx,
+            dy,
+            dt);
+
+        if(step % 100 == 0) // even steps will have currBufHost and PCurr pointing to same buffer
+        {
+            alpaka::memcpy(queue2, uBufHost, uCurrBufAcc);
+            alpaka::wait(queue2);
+            writeImage(step - 1, uBufHost, extent);
+        }
+
+        // So we just swap next to curr (shallow copy)
+        alpaka::wait(queue1);
+        std::swap(uNextBufAcc, uCurrBufAcc);
+    }
+
+    // Copy device -> host
+    alpaka::memcpy(queue1, uBufHost, uCurrBufAcc);
+    alpaka::wait(queue1);
+
+    // Validate
+    auto const [resultCorrect, maxError] = validateSolution(uBufHost, extent, dx, dy, tMax);
```

Comment:
```suggestion
    auto const [resultIsCorrect, maxError] = validateSolution(uBufHost, extent, dx, dy, tMax);
```
means that the if condition reads like fluent English.

Reply:
Done